server/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java

```diff
@@ -91,14 +97,22 @@ public SliceBuilder(String field, int id, int max) {
     }
 
     public SliceBuilder(StreamInput in) throws IOException {
-        this.field = in.readString();
+        String field = in.readString();
+        if (UidFieldMapper.NAME.equals(field) && in.getVersion().before(Version.V_6_3_0)) {
```

Comment:
I wonder if this is safe?

Like, if you slice on `_uid` in a mixed version cluster, will this cause you to get duplicate documents and to miss documents?

Reply:
I see. You've made it safe below. I think that is worth a comment.

Reply:
Will do.